spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/DiscoveryClientServiceInstanceListSupplier.java

```diff
@@ -31,19 +33,28 @@
  *
  * @author Spencer Gibb
  * @author Olga Maciaszek-Sharma
+ * @author Tim Ysewyn
  * @since 2.2.0
  */
 public class DiscoveryClientServiceInstanceListSupplier
 		implements ServiceInstanceListSupplier {
 
-	private final DiscoveryClient delegate;
-
 	private final String serviceId;
 
+	private final Flux<ServiceInstance> serviceInstances;
+
 	public DiscoveryClientServiceInstanceListSupplier(DiscoveryClient delegate,
```

Comment:
`@Deprecated`

Reply:
I'm not sure we can deprecate this. Non-reactive applications need to have support for this too. At this moment the reactive discovery client will only be enabled if webflux is on the classpath and if `spring.cloud.discovery.reactive.enabled=true` has been set (which is also the default).